assets/alertmanager/pod-disruption-budget.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  labels:
+    app.kubernetes.io/component: alert-router
+    app.kubernetes.io/instance: main
+    app.kubernetes.io/name: alertmanager
+    app.kubernetes.io/part-of: openshift-monitoring
+    app.kubernetes.io/version: 0.22.2
+  name: alertmanager-main
+  namespace: openshift-monitoring
+spec:
+  maxUnavailable: 1
```

Comment:
`maxUnavailable` is fine, maybe `minAvailable` is better
```
# oc -n openshift-monitoring get pdb 
NAME                 MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
alertmanager-main    N/A             1                 1                     39m
prometheus-adapter   1               N/A               1                     50m
prometheus-k8s       1               N/A               1                     39m
thanos-querier-pdb   1               N/A               1                     38m
```

Reply:
`maxUnavailable` comes from upstream (https://github.com/prometheus-operator/kube-prometheus/blob/6d013d4e4f980ba99cfdafa9432819d484e2f829/jsonnet/kube-prometheus/components/alertmanager.libsonnet#L154) and my understanding is that because kube-prometheus deploys 3 replicas of Alertmanager, the choice was either `maxUnavailable: 1` or `minAvailable: 2`. Ideally we should settle on one field and automatically calculate the budget.

From https://kubernetes.io/docs/tasks/run-application/configure-pdb/

> The use of maxUnavailable is recommended as it automatically responds to changes in the number of replicas of the corresponding controller.

I'll check with the workloads team if there's any strong recommendation.

Reply:
:+1: for using `maxUnavailable` here. The note about the number of replicas is relevant here imo.